### ISSUE TITLE
[REVIEW]: Mapelia and friends: create 3D models from maps

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#660]
**Submitting author:** @jordibc (<a href="http://orcid.org/0000-0002-9198-5380">Jordi Burguet-Castell</a>)
**Repository:** <a href="https://github.com/jordibc/mapelia" target ="_blank">https://github.com/jordibc/mapelia</a>
**Version:** 1.1.1
**Editor:** @arfon
**Reviewer:** @hugoledoux, @ThomasA
**Archive:** Pending

## Status

[![status](http://joss.theoj.org/papers/c13440773d37cc5abfd05016708eb351/status.svg)](http://joss.theoj.org/papers/c13440773d37cc5abfd05016708eb351)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/c13440773d37cc5abfd05016708eb351"><img src="http://joss.theoj.org/papers/c13440773d37cc5abfd05016708eb351/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/c13440773d37cc5abfd05016708eb351/status.svg)](http://joss.theoj.org/papers/c13440773d37cc5abfd05016708eb351)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/jordibc/mapelia" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@hugoledoux & @ThomasA, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @arfon know.

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jordibc/mapelia">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (1.1.1)?
- [ ] **Authorship:** Has the submitting author (@jordibc) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jordibc/mapelia">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (1.1.1)?
- [ ] **Authorship:** Has the submitting author (@jordibc) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks. @hugoledoux, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00660/joss.00660/10.21105.joss.00660.pdf)

**Comment by hugoledoux** [comment]
I am a GIS (geographical information systems) person used to work with datasets, maps, and 3D data.
So our backgrounds are very different.

## Maps?!

What is a 'map'? That term in GIS is... vague and can mean:
  1. a paper map
  2. a digital version of one, à la Google Maps

You use it to refer to gridded datasets where the value of each pixel is the elevation or something else. I would be a bit more descriptive here.

## Install

I installed it as described, with Python 3.6.5 under macOS. I assume the same would compile under Linux.

I think you should specify that Pillow and numpy are necessary in the install section, I'd have saved a few seconds.

I wonder however if it's possible under Windows? 

## version

There is no release, thus I can't confirm that v1.1.1 is there... Releases would really help in knowing what is what, now there's only the master branch.

## A statement of need

    Do the authors clearly state what problems the software is designed to solve and who the target audience is?

That is not entirely clear at this moment. Perhaps a line or two would help? What do you do with the STL files?

## Automated tests

These are not there, but I am not sure if they are needed. 

## pintelia

Why is the use of the textures a separate program? Why not just an option of mapelia?

## stl-split

"Split an stl into its north and south sides."
-->
"Split an stl into its north and south HEMISPHERES."


## paper.md

Perhaps the references to projections and formats that are in README.txt should be in here too as references?

## community guidelines

I don't see any file there.

**Comment by arfon** [comment]
:wave: @ThomasA - just a friendly reminder to try and get to this review sometime soon.

**Comment by jordibc** [comment]
Thanks @hugoledoux for the review! I found all the comments appropriate and useful. I've tried to address them:

* maps: added clarification (both in readme.rst and paper.md)
* install: added information about the necessary packages in readme.rst
* version: created a release https://github.com/jordibc/mapelia/releases/tag/v1.1.1
* a statement of need: added to paper.md an explanation of how it's used in the project A Touch of The Universe
* automated tests: actually, there were tests, but too hidden in the `examples/` directory, so I've added a link to them (`tests.py`, that can be run directly)
* pintelia: this one is in fact quite a bit different, it does not generate any kind of reliefs (only colored images superposed to a sphere) and only outputs in `ply` (because `stl` does not support colors, even if it is the format commonly used to generate 3D models for printing)
* stl-split: fixed!
* paper.md: good point too, references added
* community guidelines: added `contributing.rst` with the relevant points

Cheers,
J.

**Comment by jordibc** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Open the review issue
@whedon start review

🚧 🚧 🚧 Experimental Whedon features 🚧 🚧 🚧

# Compile the paper
@whedon generate pdf

```

**Comment by jordibc** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00660/joss.00660/10.21105.joss.00660.pdf)

**Comment by hugoledoux** [comment]
@jordibc great that you updated so fast, now all is good as far as I am concerned.

@arfon I updated my review and all the boxes are now ticked

**Comment by ThomasA** [comment]
@arfon sorry for the delay. I should be able to get to it Thursday or Friday.

**Comment by ThomasA** [comment]
I have just started reviewing the submission and I will be making remarks here as I go along. I hope that is OK. I may not be able to finish it today...

**Comment by ThomasA** [comment]
**Authorship**: It is probably as it should be, but I noticed that there are two authors, Amelia Ortiz-Gil and Jordi Burguet-Castell. On the other hand, only Jordi appears to have made [contributions](https://github.com/jordibc/mapelia/graphs/contributors) in the code repository.

**Comment by ThomasA** [comment]
**Installation instructions**: Just for those interested, the prerequisites can also easily be installed in a new Anaconda environment (here called *mapelia-review*) using:

    conda create -n mapelia-review anaconda=5 pillow numpy

Provided that libgtk is already installed.

**Comment by jordibc** [comment]
Just FYI @ThomasA , I'm totally fine with doing the remarks as you go along. I'll wait for my reply until you give me a heads up though, unless you prefer it otherwise.

**Comment by ThomasA** [comment]
@arfon, how do I make it store the ticks on my check list? If I reload the page they are gone...

**Comment by ThomasA** [comment]
**References**: I am not quite sure what the policy is concerning references, but 'paper.bib' contains two references that do not seem to be included in the paper. On the other hand, the references in the paper are not from the paper.bib file. Maybe this is the way it is intended to be, but in that case I do not see any reason to include 'paper.bib' in the repo.

**Comment by ThomasA** [comment]
**A statement of need** (documentation): I think that the documentation (in 'readme.rst') could benefit from borrowing a few lines from the paper. The paper starts out with a bit of explanation of what the package is for. I do not find that reflected in the documentation.

**Comment by ThomasA** [comment]
I think this submission meets the requirements overall and I am ready to recommend acceptance, but I would like to hear feedback on my comments on [authorship](https://github.com/openjournals/joss-reviews/issues/660#issuecomment-384677208), [references](https://github.com/openjournals/joss-reviews/issues/660#issuecomment-385200521), and [a statement of need](https://github.com/openjournals/joss-reviews/issues/660#issuecomment-385200654) first.

**Comment by jordibc** [comment]
Thanks @ThomasA ! Let me try to address those:

* Authorship: you are right, I did write the code, but Amelia got the idea, saw the directions in which it could grow, made many suggestions, tested it extensively and worked with it to generate the actual 3D-printed models --thus why she is an author too.
* References: I didn't realize the references were not used, good catch. I fixed them now. As for the ones included in `paper.md` itself, they provide context that is helpful for the work that is done with the program, but more in the category of details that I didn't consider worth discussing in the paper. Also, I find useful that they are organized in their own sections. But I am not quite sure about the policy there either.
* A statement of need: added to `readme.rst` (actually, now it reads the same as the paper).

All the changes are in the master branch. Please let me know if I should I create a tag. I suppose I should at least create one once the review is done, but I'm not sure.

Thanks also for the tip about anaconda!

Cheers,
J.

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.1240569 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.1240569" target="_blank">10.5281/zenodo.1240569</a> is the archive.
